File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let answer = 0;
    let lost_filter;
    let reserve_filter;

    lost_filter = lost.filter(e => !reserve.includes(e)).sort((a, b)=> a-b)
    reserve_filter = reserve.filter(e => !lost.includes(e)).sort((a, b)=> a-b)

    for(let i=0; i<lost_filter.length; i++) {
        for(let j=0; j<reserve_filter.length; j++) {
            if(lost_filter[i]-1 == reserve_filter[j] || lost_filter[i]+1 == reserve_filter[j]) {
                lost_filter.shift();
            }
        }
    }

    answer = n - lost_filter.length;

    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    student=[0 for i in range(n+1)]

    t1=set(lost)
    t2=set(reserve)

    lost=list(t1-t2)
    reserve=list(t2-t1)   

    c=0
    for i in lost:
        student[i]=1
        c+=1
    print(c)
    for i in reserve:
        if i>1 and student[i-1]==1:
            student[i-1]=0
            c-=1
            continue
        if i<n and student[i+1]==1:
            student[i+1]=0
            c-=1

    return n-c